File: android/app/src/main/java/github/tornaco/android/thanos/dashboard/StatusHeaderInfo.java
package github.tornaco.android.thanos.dashboard;

public class StatusHeaderInfo {
    private int runningAppsCount;
    private int memUsagePercent;
    private String memAvailablePercentString;

    StatusHeaderInfo(int runningAppsCount, int memUsagePercent, String memAvailablePercentString) {
        this.runningAppsCount = runningAppsCount;
        this.memUsagePercent = memUsagePercent;
        this.memAvailablePercentString = memAvailablePercentString;
    }

    public static StatusHeaderInfoBuilder builder() {
        return new StatusHeaderInfoBuilder();
    }

    public int getRunningAppsCount() {
        return this.runningAppsCount;
    }

    public int getMemUsagePercent() {
        return this.memUsagePercent;
    }

    public String getMemAvailablePercentString() {
        return this.memAvailablePercentString;
    }

    public static class StatusHeaderInfoBuilder {
        private int runningAppsCount;
        private int memUsagePercent;
        private String memAvailablePercentString;

        StatusHeaderInfoBuilder() {
        }

        public StatusHeaderInfo.StatusHeaderInfoBuilder runningAppsCount(int runningAppsCount) {
            this.runningAppsCount = runningAppsCount;
            return this;
        }

        public StatusHeaderInfo.StatusHeaderInfoBuilder memUsagePercent(int memUsagePercent) {
            this.memUsagePercent = memUsagePercent;
            return this;
        }

        public StatusHeaderInfo.StatusHeaderInfoBuilder memAvailablePercentString(String memAvailablePercentString) {
            this.memAvailablePercentString = memAvailablePercentString;
            return this;
        }

        public StatusHeaderInfo build() {
            return new StatusHeaderInfo(runningAppsCount, memUsagePercent, memAvailablePercentString);
        }

        public String toString() {
            return "StatusHeaderInfo.StatusHeaderInfoBuilder(runningAppsCount=" + this.runningAppsCount + ", memUsagePercent=" + this.memUsagePercent + ", memAvailablePercentString=" + this.memAvailablePercentString + ")";
        }
    }
}

File: android/app/src/main/java/now/fortuitous/thanos/dashboard/StatusHeaderInfo.kt
/*
 * (C) Copyright 2022 Thanox
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package now.fortuitous.thanos.dashboard

import github.tornaco.android.thanos.core.pm.AppInfo

data class StatusHeaderInfo(
    val runningAppsCount: Int,
    val memory: MemUsage,
    val swap: MemUsage,
    val cpu: CpuUsage
)

data class MemUsage(
    val memType: MemType = MemType.MEMORY,
    val memTotalSizeString: String = "10240M",
    // 0-100
    val memUsagePercent: Int = 88,
    val memUsageSizeString: String = "1024M",
    val memAvailableSizeString: String = "1024M",
    val isEnabled: Boolean = true
)

data class CpuUsage(
    // 0-100
    val totalPercent: Int = 1,
    val topNPkgs: List<AppCpuUsage> = emptyList()
)

data class AppCpuUsage(
    val appInfo: AppInfo,
    val percent: String,
)

enum class MemType {
    MEMORY,
    SWAP
}

val defaultStatusHeaderInfo = StatusHeaderInfo(
    999,
    MemUsage(MemType.MEMORY),
    MemUsage(MemType.SWAP),
    CpuUsage(88)
)